--- origsrc/rpy-1.0.1/src/rpymodule.c	2007-11-29 18:40:14.000000000 -0600
+++ src/rpy-1.0.1/src/rpymodule.c	2007-12-30 01:22:06.359375000 -0600
@@ -1500,7 +1500,7 @@
 PyTypeObject Robj_Type = {
   /* The ob_type field must be initialized in the module init function
    * to be portable to Windows without using C++. */
-#if defined(PRE_2_2) || defined(_WIN32)    // Matjaz
+#if defined(PRE_2_2) || defined(_WIN32) || defined(__CYGWIN__)    // Matjaz
   PyObject_HEAD_INIT(NULL)
 #else
   PyObject_HEAD_INIT(&PyType_Type)
